src/extensions/extensionInterface.ts

```diff
@@ -88,8 +85,43 @@ type RemoveLabelSortingCurationAction = {
     unitId: number
     label: string
 }
-
-export type SortingCurationAction = SetCurationSortingCurationAction | AddLabelSortingCurationAction | RemoveLabelSortingCurationAction
+type MergeUnitsSortingCurationAction = {
+    type: 'MergeUnits',
+    unitIds: number[]
+}
+type UnmergeUnitsSortingCurationAction = {
+    type: 'UnmergeUnits',
+    unitIds: number[]
+}
+
+export type SortingCurationAction = SetCurationSortingCurationAction | AddLabelSortingCurationAction | RemoveLabelSortingCurationAction | MergeUnitsSortingCurationAction | UnmergeUnitsSortingCurationAction
+
+const intersection = (a: number[], b: number[]) => (
+    a.filter(x => (b.includes(x)))
+)
+const union = (a: number[], b: number[]) => (
+    [...a, ...b.filter(x => (!a.includes(x)))].sort()
+)
+
+const simplifyMergeGroups = (mg: (number[])[]): (number[])[] => {
+    const newMergeGroups = mg.map(g => [...g]) // make a copy
+    let somethingChanged = true
+    while (somethingChanged) {
+        somethingChanged = false
+        for (let i = 0; i < newMergeGroups.length; i ++) {
+            const g1 = newMergeGroups[i]
+            for (let j = i + 1; j < newMergeGroups.length; j ++) {
+                const g2 = newMergeGroups[j]
+                if (intersection(g1, g2).length > 0) {
+                    newMergeGroups[i] = union(g1, g2)
+                    newMergeGroups[j] = []
+                    somethingChanged = true
+                }
+            }
+        }
+    }
```

Comment:
Ah, I see. You need to do union between any pair of merge groups which are not disjoint, then flatten the resulting groups, then remove any group that has fewer than 2 members (since it isn't a party if nobody comes).

Reply:
I wonder if the merge/unmerge operations would be simplified by representing the merge groups as an object whose keys are the unit IDs and whose values are Sets. Initially each object points to a Set that contains only itself. On merge, use the smallest unit ID as the key and add the full set of merged units into this Set, then update the object pointed to by all the other unit IDs to be that one.

So if we have `merge(group1 = [1, 3, 5], group2 = [2, 4, 7])` just do:
```
const finalSet = mergeGroups(group1[0]) // they all point to the same object so any one of them will do

for (const item of group2 {
  finalSet.add(item) // redundant adds are ignored
  mergeGroups[item] = finalSet // update group2's units' references to point to the final set, which will be mutated in place
}
```

And then to unmerge:
```
for (const x in items_to_unmerge) {
  mergeGroups[x].remove(x)
  mergeGroups[x] = new Set([x])
}
```

That said, what you've already written is already in place and working, and I don't know about JSON serializability (if we'd need to convert the Sets back to lists or something). Initial state is easy when nothing is merged, but loading a curation that has some merges in it would be non-obvious as you'd have to compare the lists and make sure they all pointed to the correct Set reference (rather than initializing with multiple Sets that happen to have the same values).

Reply:
while there are advantages to what you propose in terms of ease of manipulation of the merge groups, and access, I think I prefer to have a simpler representation (sets of disjoint sets) and then create convenience functions for doing the manipulations. We're not worried about efficiency here.